### DISCUSSION TITLE
Question about Setting Global CH4 Emission Constraint under SSP245 in GCAM

### DISCUSSION THREAD

**Discussion by jiangyongye** [JGCRI/gcam-core#282] [Q&A] [answered]
Could anyone pls help me ^ _ ^'🥺🙏
I want to set a constraint X for global CH4 emission in 2030 under SSP245, which means global can only have X amount of CH4 emission in 2030 under SSP245.
That's what I tried.
First, as far as I know, to be able to run SSP245 in gcam6, we can only use configuration_policy.xml as a configuration file in target finder mode, and the "policy-target-file" should be the provided 'forcing_target_4p5.xml'. (I only get the correct ssp245 output in this way, and never in normal mode) Correct me if you have another way to output the correct ssp245 : ( I think the original configuration.xml in normal mode outputs the SSP2 baseline. 
So I think if I want to run my 2030CH4 scenario, I just need to add additional some stuff in my ssp245 target finder setting.
The first thing I tried is to create a constraint XML and added XML file name into <ScenarioComponents> of the configuration file.
The constraint XML I add is like this :
<img width="468" alt="1682426330650" src="https://user-images.githubusercontent.com/49706086/234278449-01c6bdaf-3ab0-49db-9019-2f5dde3b12b7.png">
but it does not work, looks like gcam can only allow user to set constraints for a specific region not global.

The second thing I tried is added some additional lines in the 'forcing_target_4p5.xml', so I want to gcam can find a path that achieves both targets: 1: global co2 concentration 4.5 at 2100, 2: global CH4 emission x at 2030
so my  'forcing_target_4p5.xml' are:
![image](https://user-images.githubusercontent.com/49706086/234278594-415822e9-fc2f-4e70-90a2-8eb69ddc225d.png)
However, It still does not work.

Could anyone pls help me🥺🙏

**Comment by jiangyongye** [comment, marked as answer]
Nvm
I figured it out. In case someone has the same question, the constraint XML file should be like this:
![image](https://user-images.githubusercontent.com/49706086/234756807-cc927860-0de8-4d5b-b87f-2b6017211c8f.png)
all the regions share the same market will create a global CH4 emission constraint

**Comment by ssmithClimate** [comment]
Glad to hear! A couple notes.

The MAC curves in GCAM are parametrized to respond to GCAM's carbon prices. You can use them to respond to a CH4 price, but you need to change the MAC curve XMLs to work with these units.

Note that one decision you will also need to also make is if you want to configure this as: 1) CH4 reductions are only through MAC curves, or 2) have the CH4 price feed into economic decision making at the technology level (You can do this separately for agriculture vs other CH4 emissions) using the `price-adjust` parameter in the linked market setup. See the [GCAM documentation](https://jgcri.github.io/gcam-doc/emissions.html) for more details.
